ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/tests/ContractKeys.scala

```diff
@@ -162,10 +163,22 @@ final class ContractKeys(session: LedgerSession) extends LedgerTestSuite(session
       }
     }
 
+  val recreateContractKeys =
+    LedgerTest("CKRecreate", "Contract keys can be recreated in single transaction") { context =>
+      val key = s"${UUID.randomUUID.toString}-key"
+      for {
+        ledger <- context.participant()
+        Vector(owner) <- ledger.allocateParties(1)
+        delegated <- ledger.create(owner, Delegated(owner, key))
+        _ <- ledger.exercise(owner, delegated.exerciseRecreate(_))
+      } yield { () }
```

Comment:
I think we can still make some assertions here:
for example:
asserting that the contract key from the original `Delegated` contract is the same as the contract key from the replacement, but the contract IDs themselves are different.

Reply:
thanks, good point. will add the assertions.